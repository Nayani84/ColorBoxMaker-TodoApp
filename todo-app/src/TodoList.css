.Todo-list {
    max-width: 500px;
    margin: 0 auto;
    text-align: left;
}

.Todo-container {
    margin-top: 20px;
    border: 1px solid #ddd;
    padding: 10px;
    border-radius: 5px;
}