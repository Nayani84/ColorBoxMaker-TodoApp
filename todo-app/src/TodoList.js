import React, { useState } from "react";
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import './TodoList.css';

const TodoList = () => {
    const INITIAL_STATE = [];
    const [todos, setTodos] = useState(INITIAL_STATE);
    const addTodo = (task) => {
        setTodos(todos => [...todos, task]);
    };

    const updateTodo = (id, editTask) => {
        setTodos(todos => todos.map(todo => todo.id === id ? { ...todo, task: editTask } : todo));
    };

    const removeTodo = (id) => {
        setTodos((todos) => todos.filter((todo) => todo.id !== id));
    };


    return (
        <div className="Todo-list">
            <NewTodoForm addTodo={addTodo} />
            <div className="Todo-container">
                {todos.map((todo) => (
                    <Todo
                        key={todo.id}
                        id={todo.id}
                        task={todo.task}
                        updateTodo={updateTodo}
                        removeTodo={removeTodo}
                    />
                ))}
            </div>
        </div>
    );
}

export default TodoList;