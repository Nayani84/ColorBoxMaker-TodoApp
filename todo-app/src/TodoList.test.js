import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

// Smoke test
it("renders without crashing", function () {
    render(<TodoList />);
});

// Snapshot test 
it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});


it("should add new task", function () {
    const { queryByText, getByLabelText } = render(<TodoList/>);

    const input = getByLabelText("Task :");
    const btn = queryByText("Add Task");

    expect(queryByText("X")).not.toBeInTheDocument();

    fireEvent.change(input, { target: { value: "Test todo" } });
    fireEvent.click(btn);

    expect(queryByText('X')).toBeInTheDocument();
    expect(queryByText("Test todo")).toBeInTheDocument();
});

it("should update task", function () {
    const { queryByText, getByLabelText, getByDisplayValue, getByRole } = render(<TodoList/>);

    const input = getByLabelText("Task :");
    const btn = queryByText("Add Task");

    expect(queryByText("Edit")).not.toBeInTheDocument();

    fireEvent.change(input, { target: { value: "Test Task" } });
    fireEvent.click(btn);

    const editButton = queryByText("Edit");
    expect(editButton).toBeInTheDocument();
    expect(queryByText("Test Task")).toBeInTheDocument();
    fireEvent.click(editButton);

    
    const editInput = getByDisplayValue("Test Task");
    fireEvent.change(editInput, { target: { value: "Updated task" } });
    const saveButton = queryByText("Save"); 
    fireEvent.click(saveButton);

    expect(queryByText("Updated task")).toBeInTheDocument();
    expect(queryByText("Test Task")).not.toBeInTheDocument();
});

it("should remove task", function () {
    const { queryByText, getByLabelText } = render(<TodoList/>);

    const input = getByLabelText("Task :");
    const btn = queryByText("Add Task");

    expect(queryByText("X")).not.toBeInTheDocument();

    fireEvent.change(input, { target: { value: "Test todo" } });
    fireEvent.click(btn);

    const removeButton = queryByText("X");
    expect(removeButton).toBeInTheDocument();
    expect(queryByText("Test todo")).toBeInTheDocument();

    fireEvent.click(removeButton);
    expect(removeButton).not.toBeInTheDocument();
    expect(queryByText("Test todo")).not.toBeInTheDocument();
});