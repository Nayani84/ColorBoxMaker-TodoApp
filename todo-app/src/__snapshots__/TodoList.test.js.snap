// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo-list"
  >
    <form
      class="New-todo-form"
    >
      <label
        for="task"
      >
        Task :
      </label>
      <input
        id="task"
        name="task"
        placeholder="Task"
        type="text"
        value=""
      />
      <button>
        Add Task
      </button>
    </form>
    <div
      class="Todo-container"
    />
  </div>
</DocumentFragment>
`;
